docs/building-modules/module-interfaces.md

```diff
@@ -6,202 +6,134 @@ order: 11
 
-In order to do that, modules should implement `RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux)` on `AppModuleBasic` to wire the client gRPC requests to the correct handler inside the module.
+In order to do that, modules must implement `RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux)` on `AppModuleBasic` to wire the client gRPC requests to the correct handler inside the module.
 
-Here's an example from the `auth` module:
+Here's an example from the `x/auth` module:
 
-+++ https://github.com/cosmos/cosmos-sdk/blob/64b6bb5270e1a3b688c2d98a8f481ae04bb713ca/x/auth/module.go#L69-L72
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.43.0-beta1/x/auth/module.go#L68-L73
 
 ## gRPC-gateway REST
 
-Applications typically support web services that use HTTP requests (e.g. a web wallet like [Lunie.io](https://lunie.io)). Thus, application developers can also use REST Routes to route HTTP requests to the application's modules; these routes will be used by service providers.
-
-[grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) translates REST calls into gRPC calls, which might be useful for clients that do not use gRPC.
+Applications need to support web services that use HTTP requests (e.g. a web wallet like [Keplr](https://keplr.xyz)). [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) translates REST calls into gRPC calls, which might be useful for clients that do not use gRPC.
 
-Modules that want to expose REST queries should add `google.api.http` annotations to their `rpc` methods, such as in the example below from the `auth` module:
+Modules that want to expose REST queries should add `google.api.http` annotations to their `rpc` methods, such as in the example below from the `x/auth` module:
 
-```proto
-// Query defines the gRPC querier service.
-service Query{
-    // Account returns account details based on address.
-    rpc Account (QueryAccountRequest) returns (QueryAccountResponse) {
-      option (google.api.http).get = "/cosmos/auth/v1beta1/accounts/{address}";
-    }
-
-    // Params queries all parameters.
-    rpc Params (QueryParamsRequest) returns (QueryParamsResponse) {
-      option (google.api.http).get = "/cosmos/auth/v1beta1/params";
-    }
-}
-```
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.43.0-beta1/proto/cosmos/auth/v1beta1/query.proto#L13-L29
 
 gRPC gateway is started in-process along with the application and Tendermint. It can be enabled or disabled by setting gRPC Configuration `enable` in [`app.toml`](../run-node/run-node.md#configuring-the-node-using-apptoml).
 
 The SDK provides a command for generating [Swagger](https://swagger.io/) documentation (`protoc-gen-swagger`). Setting `swagger` in [`app.toml`](../run-node/run-node.md#configuring-the-node-using-apptoml) defines if swagger documentation should be automatically registered.
 
-## Legacy REST
```

Comment:
Why was this removed?

Reply:
I removed this section because new modules should not be built using the old pattern and this section explains how to create a REST interface using the old pattern.

As far as I understand, developers should use gRPC-gateway to create REST endpoints moving forward. I still think we need to provide information about [Legacy REST](https://docs.cosmos.network/v0.42/core/grpc_rest.html#rest-server) and how to [migrate to gRPC](https://docs.cosmos.network/v0.42/migrations/rest.html#rest-endpoints-migration), but explaining how to build a REST interface for new modules seemed unnecessary.

Maybe some of this information should be moved rather than removed? I would be happy to fix.

Reply:
ah okay, makes sense.

Reply:
I think it's fine to remove it, in any case it'll be still accessible from previous versions of the docs (eg https://docs.cosmos.network/v0.39/building-modules/module-interfaces.html#rest)